src/client/public/scripts/dash/auto-merge-events.ts

```diff
@@ -0,0 +1,165 @@
+import {html} from '../../../../../node_modules/lit-html/lib/lit-extended.js';
+import {TemplateResult} from '../../../../../node_modules/lit-html/lit-html.js';
+import * as api from '../../../../types/api.js';
+
+import {EventModel, eventTemplate} from './pr-event.js';
+
+// TODO: This should be self contained to each auto-merge instance, not
+// a global for all auto-merge instances
+const automergeOpenStates: {[prId: string]: boolean} = {};
+
+function selectAutomergeOpt(
+    pr: api.OutgoingPullRequest,
+    optionSelected: 'manual'|'merge'|'squash'|'rebase',
+) {
+  // TODO: Would be good to mark this change as disabled until the network
+  // request has finished.
+  fetch('/api/auto-merge/set-merge-option/', {
+    method: 'POST',
+    credentials: 'include',
+    body: JSON.stringify({
+      prId: pr.id,
+      automergeOption: optionSelected,
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+    }
+  }).then(async (response) => {
+    if (!response.ok) {
+      const reponseText = await response.text();
+      console.error('Unable to set auto merge option', reponseText);
+      return;
+    }
+  });
+
+  if (!pr.automergeOpts) {
+    pr.automergeOpts = {
+      mergeType: optionSelected,
+    };
+  } else {
+    pr.automergeOpts.mergeType = optionSelected;
+  }
+
+  // Force close
+  automergeOpenStates[pr.id] = false;
+  requestRender();
+}
+
+function requestRender() {
+  window.postMessage(
+      {
+        type: 'render-request',
+      },
+      window.location.origin,
+  );
+}
+
+export function getAutoMergeOptions(pr: api.PullRequest): TemplateResult[] {
+  if (!('mergeable' in pr)) {
+    // This is not an outgoing PR
+    return [];
+  }
+
+  const outgoingPr = (pr as api.OutgoingPullRequest);
```

Comment:
Not a fan of this. We already knew the type of this, so it doesn't seem great to then merge and split it out again. Don't have a good idea of